test/integration/auth.spec.ts

```diff
@@ -0,0 +1,356 @@
+/*!
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as admin from '../../lib/index';
+import {expect} from 'chai';
+import * as chai from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+import firebase = require('firebase');
+import {clone} from 'lodash';
+import {generateRandomString, projectId, apiKey} from './setup';
+
+chai.should();
+chai.use(chaiAsPromised);
+
+const newUserUid = generateRandomString(20);
+const nonexistentUid = generateRandomString(20);
+const testPhoneNumber = '+11234567890';
+const testPhoneNumber2 = '+16505550101';
+const nonexistentPhoneNumber = '+18888888888';
+const updatedEmail = generateRandomString(20) + '@example.com';
+const updatedPhone = '+16505550102';
+const customClaims = {
+  admin: true,
+  groupId: '1234',
+};
+const uids = [newUserUid + '-1', newUserUid + '-2', newUserUid + '-3'];
+const mockUserData = {
+  email: newUserUid + '@example.com',
+  emailVerified: false,
+  phoneNumber: testPhoneNumber,
+  password: 'password',
+  displayName: 'Random User ' + newUserUid,
+  photoURL: 'http://www.example.com/' + newUserUid + '/photo.png',
+  disabled: false,
+};
+
+describe('admin.auth', () => {
+
+  let uidFromCreateUserWithoutUid: string;
+
+  before(() => {
+    firebase.initializeApp({
+      apiKey,
+      authDomain: projectId + '.firebaseapp.com',
+    });
+    cleanup();
```

Comment:
I'm not sure this cleanup is necessary.  (the tests are the same * see other comment) without it.

Reply:
This is required since we are creating users with fixed phone numbers. This ensures that we clean up those users (created during a previous incomplete execution of the suite) before starting a new run.

Reply:
I couldn't get tests to fail without it, (but perhaps this is due to the lucky order of the runs.)
LGTM